src/audio/dai.c

```diff
@@ -691,6 +691,27 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 	return 0;
 }
 
+static void dai_cache(struct comp_dev *dev, int cmd)
+{
+	struct dai_data *dd = comp_get_drvdata(dev);
+	cache_command cache_cmd = comp_get_cache_command(cmd);
+	struct list_item *item;
+	struct dma_sg_elem *e;
+
+	trace_dai("cac");
+
+	list_for_item(item, &dd->config.elem_list) {
```

Comment:
The sequence for invalidation may be different here as we first need to invalidate the list head beofre we can iterate over it. We should probably lock this to synchronize across different cores ?